File: Day 18-21/memory.js
function random(min, max)
{
    var res;
    res = Math.floor(Math.random() * (max - min +1)) + min;
    return res;
}

const chars = `#$%()/&"`

function getNRandomChars(text, n){

    textOut = ""

    for( let i=0; i<n; i++){
        let randomChar = text.charAt(random(0,text.length))
        textOut+= randomChar
    }
    console.log("Remember this text:")
    console.log(textOut)

    return textOut
}

function requestNChars(n){

    console.clear()

    userText = ""

    for( let i=0; i<n; i++){
        let userChar = prompt(`Type a char of theese: ${chars}`)
        if(chars.includes(userChar))
            userText+= userChar
        else{
            i--
            alert("This character is not valid")
        }
    }
    console.log(`Your text is ${userText}`)

    return userText
}

randomText = getNRandomChars(chars, 4)

setTimeout( function(){
    
    userText = requestNChars(4)

    if( randomText === userText)
        console.log("Your sequence is correct! Congratulations!")
    else
        console.log("Your sequence is wrong! Sorry!")

}, 8000)

File: Day 18-21/memory.cpp
#include <iostream>
#include <math.h>
#include <string>
#include <stdlib.h>
#include<unistd.h>

using namespace std;

char symbols[8] = {'#','$','%','(',')','/','&','"'};
string randomChars(char chars[], int len);
string userChars(int len);
bool isCharInSymbols(char ch);

int main()
{

    int TEXT_SIZE = 4;
    string randomText = randomChars(symbols, TEXT_SIZE);
    string userText = userChars(TEXT_SIZE);

    if( randomText == userText )
        printf("Your sequence is correct! Congratulations!");
    else
        printf("Your sequence is wrong! Try it again!");

    return 0;
}

string randomChars(char chars[], int len){

    char textOut[len];

    int i;
    for( i=0; i<len; i++){

        int number = rand()% len;
        char randomCh = chars[number];
        textOut[i] = randomCh; 
    }

    cout << "Try to repeat this sequence: " << endl;
    cout << textOut;

    sleep(5);

    int j;
    for(j=0; j<len; j++){
        printf("\b");
    }

    return textOut;
}

string userChars(int len){

    char userText[len];

    int i;
    for( i=0; i<len; i++){
        cout << "Type a character include in ";
        int j;
        for( j=0; j<8; j++){
            cout << symbols[j];
        }
        cout << endl;
        cin >> userText[i];

        if( !isCharInSymbols(userText[i]) ){
            cout << "This character is not valid" << endl;
            i--;
        }

    }

    userText[len] = '\0';
    cout << userText << endl;

    return userText;

}


bool isCharInSymbols(char ch){

    int i;
    for( i=0; i < sizeof(symbols)/sizeof(symbols[0]); i++)
    {
        if ( i == sizeof(symbols) - 1 )
             return false;
        if( ch == symbols[i] )
            return true;

    }
}